src/analysis/inner_product_space/calculus.lean

```diff
@@ -240,3 +240,101 @@ lemma differentiable_on.dist (hf : differentiable_on ℝ f s) (hg : differentiab
   (hne : ∀ x ∈ s, f x ≠ g x) :
   differentiable_on ℝ (λ y, dist (f y) (g y)) s :=
 λ x hx, (hf x hx).dist (hg x hx) (hne x hx)
+
+section pi_like
+
+open continuous_linear_map
+
+#where
+
+variables {H : Type*} [normed_group H] [normed_space 𝕜 H] {ι : Type*} [fintype ι] {t : set H}
+  {y : H}
+
+lemma differentiable_within_at_euclidean {f : H → euclidean_space 𝕜 ι} :
+  differentiable_within_at 𝕜 f t y ↔ ∀ i, differentiable_within_at 𝕜 (λ x, f x i) t y :=
+begin
+  let φ : euclidean_space 𝕜 ι ≃L[𝕜] ι → 𝕜 := (linear_equiv.refl _ _).to_continuous_linear_equiv,
+  rw [← φ.comp_differentiable_within_at_iff, differentiable_within_at_pi],
+  refl
+end
+
+lemma differentiable_at_euclidean {ι : Type*} [fintype ι] {f : H → euclidean_space 𝕜 ι} :
```

Comment:
You have `{ι : Type*} [fintype ι]` already in context.  Also, can you extract `{f : H → euclidean_space 𝕜 ι}` into the `variables` for use throughout the section?